x-pack/plugins/security_solution/server/lib/detection_engine/signals/utils.ts

```diff
@@ -164,81 +164,21 @@ export const checkPrivileges = async (
   });
 
 export const getGapMaxCatchupRatio = ({
-  logger,
-  previousStartedAt,
-  unit,
-  buildRuleMessage,
-  ruleParamsFrom,
-  interval,
+  gap,
+  intervalDuration,
 }: {
-  logger: Logger;
-  ruleParamsFrom: string;
-  previousStartedAt: Date | null | undefined;
-  interval: string;
-  buildRuleMessage: BuildRuleMessage;
-  unit: string;
-}): {
-  maxCatchup: number | null;
-  ratio: number | null;
-  gapDiffInUnits: number | null;
-} => {
-  if (previousStartedAt == null) {
-    return {
-      maxCatchup: null,
-      ratio: null,
-      gapDiffInUnits: null,
-    };
-  }
-  if (!isValidUnit(unit)) {
-    logger.error(buildRuleMessage(`unit: ${unit} failed isValidUnit check`));
-    return {
-      maxCatchup: null,
-      ratio: null,
-      gapDiffInUnits: null,
-    };
-  }
-  /*
-      we need the total duration from now until the last time the rule ran.
-      the next few lines can be summed up as calculating
-      "how many second | minutes | hours have passed since the last time this ran?"
-      */
-  const nowToGapDiff = moment.duration(moment().diff(previousStartedAt));
-  // rule ran early, no gap
-  if (shorthandMap[unit].asFn(nowToGapDiff) < 0) {
-    // rule ran early, no gap
-    return {
-      maxCatchup: null,
-      ratio: null,
-      gapDiffInUnits: null,
-    };
-  }
-  const calculatedFrom = `now-${
-    parseInt(shorthandMap[unit].asFn(nowToGapDiff).toString(), 10) + unit
-  }`;
-  logger.debug(buildRuleMessage(`calculatedFrom: ${calculatedFrom}`));
-
-  const intervalMoment = moment.duration(parseInt(interval, 10), unit);
-  logger.debug(buildRuleMessage(`intervalMoment: ${shorthandMap[unit].asFn(intervalMoment)}`));
-  const calculatedFromAsMoment = dateMath.parse(calculatedFrom);
-  const dateMathRuleParamsFrom = dateMath.parse(ruleParamsFrom);
-  if (dateMathRuleParamsFrom != null && intervalMoment != null) {
-    const momentUnit = shorthandMap[unit].momentString as moment.DurationInputArg2;
-    const gapDiffInUnits = dateMathRuleParamsFrom.diff(calculatedFromAsMoment, momentUnit);
-
-    const ratio = gapDiffInUnits / shorthandMap[unit].asFn(intervalMoment);
-
-    // maxCatchup is to ensure we are not trying to catch up too far back.
-    // This allows for a maximum of 4 consecutive rule execution misses
-    // to be included in the number of signals generated.
-    const maxCatchup = ratio < MAX_RULE_GAP_RATIO ? ratio : MAX_RULE_GAP_RATIO;
-    return { maxCatchup, ratio, gapDiffInUnits };
+  gap: moment.Duration;
+  intervalDuration: moment.Duration;
+}): number => {
+  const gapInMilliseconds = gap.asMilliseconds();
+  if (gapInMilliseconds <= 0) {
+    return 0;
   }
-  logger.error(buildRuleMessage('failed to parse calculatedFrom and intervalMoment'));
-  return {
-    maxCatchup: null,
-    ratio: null,
-    gapDiffInUnits: null,
-  };
+  const ratio = Math.ceil(gapInMilliseconds / intervalDuration.asMilliseconds());
```

Comment:
Can `intervalDuration.asMilliseconds()` be `0`?

Reply:
I hope that gets prevented in the alerting framework, but it's a good point. I added a check to prevent divide by 0 here.